pkg/validation/runner.go

```diff
@@ -0,0 +1,143 @@
+package validation
+
+import (
+	"context"
+	"reflect"
+	"sync"
+
+	"github.com/aws/eks-anywhere/pkg/errors"
+)
+
+// Validatable is anything that can be validated.
+type Validatable[O any] interface {
+	Copiable[O]
+}
+
+// Copiable is anything that can be copied.
+type Copiable[O any] interface {
+	DeepCopy() O
+}
+
+// Validation is the logic for a validation of a type O.
+type Validation[O Validatable[O]] func(ctx context.Context, obj O) error
+
+// Runner allows to compose and run validations.
+type Runner[O Validatable[O]] struct {
+	validations []Validation[O]
+	config      *RunnerConfig
+}
+
+// RunnerConfig contains the configuration for a Runner.
+type RunnerConfig struct {
+	maxJobs int
+}
+
+// RunnerOpt allows to configure a Runner with optional parameters.
+type RunnerOpt func(*RunnerConfig)
+
+// WithMaxJobs sets the maximun number of concurrent routines the runner will use.
+func WithMaxJobs(m int) RunnerOpt {
+	return func(c *RunnerConfig) {
+		c.maxJobs = m
+	}
+}
+
+// NewRunner constructs a new Runner.
+func NewRunner[O Validatable[O]](opts ...RunnerOpt) *Runner[O] {
+	r := &Runner[O]{
+		config: &RunnerConfig{
+			maxJobs: 8,
```

Comment:
Magic number.

Reply:
Defaulting to `runtime.GOMAXPROCS`

Reply:
I was thinking more about having a constant for it instead of defining in-place but on second look I don't think it's going to make a difference.